Gemfile

```diff
@@ -31,6 +31,7 @@ group :test do
   gem "test-theme-symlink", :path => File.expand_path("test/fixtures/test-theme-symlink", __dir__)
 
   gem "jruby-openssl" if RUBY_ENGINE == "jruby"
+  gem "eventmachine", "1.2.7", git: "https://github.com/eventmachine/eventmachine", tag: "v1.2.7" if Gem.win_platform?
```

Comment:
What is the difference between this and using the released version (`v1.2.7`) directly?

Reply:
If done without this approach (by just listing `gem "eventmachine"` in the Gemfile), the package will be downloaded and built locally as EventMachine-mingw32. The native extension versions does not work in windows. It builds/installs, but fails in use.  This is why you've had to comment out the tests in Windows in the past. 

It is not about the specific version, it is about getting a pure ruby version of the gem. The eventmachine gem will work on windows.  It just has to be the version built with --platform ruby.  Unfortunately trying to set the platform to ruby using bundler or in the Gemfile is still problematic.  See https://github.com/rubygems/rubygems/issues/3347. 

The approach used here causes the gem to be built as if this were done an the command line: `gem install eventmachine --platform ruby `. The only difference is a bit more network traffic, since there isn't a locally built copy - the gem is pulled from github.

```
$ bundle install | grep eventmachine
Using eventmachine 1.2.7 from https://github.com/eventmachine/eventmachine (at v1.2.7@9cb13a8)
```

Someone else came up with this incantation (https://www.dixis.com/posts/171). It was a good find that helps solve/work-around this issue until such time as bundler and Gemfiles can accommodate specifying --platform ruby.

It would also be possible to manually install  ```gem install eventmachine --platform ruby``` and then update the Gemfile to use eventmachine. But that requires manual work, and would not be something that would work for the automated tests.